app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -280,6 +280,12 @@ class StateFragmentPresenter @Inject constructor(
     }
 
     val ephemeralState = result.getOrThrow()
+
+    var scrollToTop = false
+    if (currentStateName.isNotEmpty() && currentStateName != ephemeralState.state.name) {
```

Comment:
Let's keep currentStateName a lateinit var. You can check if it's initialized, which I think devolves to something like:

```kotlin
val scrollToTop = ::currentStateName.isInitialized && currentStateName != ephemeralState.state.name
```

Also note that this simplifies scrollToTop and allows it to be a val.

Reply:
Done

Reply:
Done